sql/telemetry_derived/clients_daily_histogram_aggregates_v1.sql.py

```diff
@@ -43,49 +43,25 @@
 
 
 def generate_sql(opts, additional_queries, windowed_clause, select_clause, json_output):
-    """Create a SQL query for the clients_daily_histogram_aggregates dataset."""
-    get_keyval_pairs = """
-        CREATE TEMPORARY FUNCTION get_keyval_pairs(y STRING)
-        RETURNS ARRAY<STRING>
-        LANGUAGE js AS
-        '''
-          var z = new Array();
-          node = JSON.parse(y);
-          for (let [key, value] of Object.entries(node)) {
-            z.push(`${key}:${value}`);
-          }
-          return z;
-        ''';
-    """
-
-    string_to_arr = """
-        -- convert a string like '[5, 6, 7]' to an array struct
-        CREATE TEMPORARY FUNCTION string_to_arr(y STRING)
-        RETURNS ARRAY<STRING>
-        LANGUAGE js AS
-        '''
-          return JSON.parse(y);
-        ''';
-    """
-
     """Create a SQL query for the clients_daily_histogram_aggregates dataset."""
     query = textwrap.dedent(
-        f"""-- Query generated by: templates/clients_daily_histogram_aggregates.sql.py
-        {get_keyval_pairs}
-
-        {string_to_arr}
-
-        CREATE TEMP FUNCTION
-          udf_aggregate_json_sum(histograms ARRAY<STRING>) AS (ARRAY(
-              SELECT
-                AS STRUCT SPLIT(keyval, ':')[OFFSET(0)] AS key,
-                SUM(SAFE_CAST(SPLIT(keyval, ':')[OFFSET(1)] AS INT64)) AS value
-              FROM
-                UNNEST(histograms) AS histogram,
-                UNNEST(get_keyval_pairs(JSON_EXTRACT(histogram, "$.values"))) AS keyval
-              WHERE histogram IS NOT NULL
-                AND JSON_EXTRACT(histogram, "$.values") IS NOT NULL
-              GROUP BY key));
+        f"""-- Query generated by: sql/telemetry_derived/clients_daily_histogram_aggregates_v1.sql.py --agg-type {opts["agg_type"]}
+        CREATE TEMP FUNCTION udf_aggregate_json_sum(histograms ARRAY<STRING>) AS (
+          ARRAY(
+            SELECT AS STRUCT
+              FORMAT('%d', values_entry.key) AS key,
+              SUM(values_entry.value) AS value
+            FROM
+              UNNEST(histograms) AS histogram,
+              UNNEST(mozfun.hist.extract(histogram).values) AS values_entry
+            WHERE
+              histogram IS NOT NULL
+            GROUP BY
+              values_entry.key
+            ORDER BY
+              values_entry.key
```

Comment:
I've rebased on master and added an ORDER BY here that appears to be necessary to get the tests to run successfully.